File: src/test/java/io/github/intellij/dlanguage/highlighting/DlangHighlightingLexerTest.java
package io.github.intellij.dlanguage.highlighting;

public class DlangHighlightingLexerTest extends DHighlightingLexerTestBase {

    public DlangHighlightingLexerTest() {
        super("highlighting");
    }

    // Ensure that the parser don’t crash if a comment is unclosed
    public void testcomment_unclosed() { doTest("/* unclosed comment", "DlangTokenType.BLOCK_COMMENT ('/* unclosed comment')", createLexer()); }

    // strings
    public void testchar() {
        doTest(true, true);
    }
    public void teststrings_escapes() {
        doTest(true, true);
    }
}

File: src/test/kotlin/io/github/intellij/dlanguage/highlighting/DlangHighlightingLexerTest.kt
package io.github.intellij.dlanguage.highlighting

class DlangHighlightingLexerTest : DHighlightingLexerTestBase("highlighting") {
    // Ensure that the parser doesn’t crash if a comment is unclosed
    fun testcomment_unclosed() {
        doTest("/* unclosed comment", "DlangTokenType.BLOCK_COMMENT ('/* unclosed comment')", createLexer())
    }

    // strings
    fun testchar() = doTest()
    fun teststrings_escapes() = doTest()
}
